tests/unit/test_scenario.py

```diff
@@ -0,0 +1,59 @@
+from unittest.mock import ANY
+
+import ops
+import pytest
+from charm import HexanatorCharm
+from scenario import Container, Context, Relation, State
+
+
+def default_pebble_layer() -> dict:
+    import yaml
+
+    tmp = yaml.safe_load(open("rockcraft.yaml").read())
+    assert tmp["services"]["gubernator"]["startup"] == "enabled"
+    return tmp
+
+
+@pytest.fixture
+def ctx():
+    return Context(HexanatorCharm)
+
+
+@pytest.fixture
+def initial_state():
+    pebble_layers = {"default": ops.pebble.Layer(raw=default_pebble_layer())}  # type: ignore
+    container = Container("gubernator", can_connect=True, layers=pebble_layers)
+    ingress = Relation("ingress", id=0, interface="ingress", remote_app_name="ingress")
+    rate_limit = Relation("rate-limit", id=1, interface="http", remote_app_name="user")
+    return State(leader=True, relations={ingress, rate_limit}, containers={container})  # type: ignore
+
+
+def test_startup(ctx, initial_state):
+    container = list(initial_state.containers)[0]
```

Comment:
We'd like to avoid people referring to state components by number (this is one of the reasons they are changing from lists to sets). You know the name of the container, so it seems simplest to just do:

```suggestion
    container = initial_state.get_container("gubernator")
```